src/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/PerformanceCounterLib.cs

```diff
@@ -835,7 +836,7 @@ private CategorySample GetCategorySample(string category)
                 return null;
 
             CategorySample sample = null;
-            byte[] dataRef = GetPerformanceData(entry.NameIndex.ToString(CultureInfo.InvariantCulture));
+            byte[] dataRef = GetPerformanceData(entry.NameIndex.ToString(CultureInfo.InvariantCulture), true);
```

Comment:
nit , named parameter is a little more readable
`byte[] dataRef = GetPerformanceData(entry.NameIndex.ToString(CultureInfo.InvariantCulture), usePool:true);`

Reply:
Or maybe just remove the bool and make all code paths use the pool.

Reply:
named parameter added. I don't think that all code paths should use the pool - we need release rented bytes - i think it is not actual for global perf counter configuration reading.